moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -1835,12 +1838,32 @@ bool PlanningScene::processCollisionObjectMove(const moveit_msgs::CollisionObjec
     }
 
     collision_detection::World::ObjectConstPtr obj = world_->getObject(object.id);
+
+    // Update subframe poses
+    moveit::core::FixedTransformsMap subframe_poses = obj->subframe_poses_;
+    if (!object.subframe_poses.empty())
+    {
+      Eigen::Isometry3d frame_pose;
+      for (std::size_t i = 0; i < object.subframe_poses.size(); ++i)
+      {
+        tf2::fromMsg(object.subframe_poses[i], frame_pose);
+        std::string name = object.subframe_names[i];
+        subframe_poses[name] = t * frame_pose;
+      }
+    }
+    else if (!obj->subframe_poses_.empty())  // If there were subframes but no new poses are given
+    {
+      ROS_WARN_NAMED(LOGNAME, "Object '%s' had subframes, but no new ones were given for Move operation. ",
+                     object.id.c_str());
+    }
+
     if (obj->shapes_.size() == new_poses.size())
     {
       std::vector<shapes::ShapeConstPtr> shapes = obj->shapes_;
       obj.reset();
       world_->removeObject(object.id);
       world_->addToObject(object.id, shapes, new_poses);
+      world_->setSubframesOfObject(object.id, subframe_poses);
```

Comment:
isn't updating the subframes logically independent of specifying poses for everything else?

i can't help but think the use case you want to allow here could be much nicer included in APPEND.
maybe we could just add subframes to the warning in the beginning that says they will be ignored?

Reply:
My first thought was the very same as Michael's here: Why should we couple an update of shapes to the updates of subframes.
On the other hand, moving shapes, typically also invalidates subframes and thus it makes sense to update the latter in the same call. The present code essentially replicates the corresponding code of the `APPEND` mode. A warning is issued when shapes have moved, but no updates (at all) were specified for subframes.

Reply:
> The present code essentially replicates the corresponding code of the APPEND mode.

I would argue the code should sanitize the subframes list and forbid adding new frames here.
After all, if you want to add new frames, you can use `APPEND`.

Reply:
I removed this because the logic will change in #2037